pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -396,63 +400,74 @@ func (c *DeploymentConfigController) calculateStatus(config deployapi.Deployment
 		deployutil.SetDeploymentCondition(newStatus, *noMinAvailability)
 	}
+
 	// Condition about progress.
 	cond := deployutil.GetDeploymentCondition(*newStatus, deployapi.DeploymentProgressing)
 	if latestRC != nil {
 		switch deployutil.DeploymentStatusFor(latestRC) {
-		case deployapi.DeploymentStatusNew, deployapi.DeploymentStatusPending:
-			msg := fmt.Sprintf("Waiting on deployer pod for %q to be scheduled", latestRC.Name)
+		case deployapi.DeploymentStatusPending:
+			msg := fmt.Sprintf("Waiting on deployer pod for replication controller %q to be scheduled", latestRC.Name)
 			condition := deployutil.NewDeploymentCondition(deployapi.DeploymentProgressing, kapi.ConditionUnknown, "", msg)
 			deployutil.SetDeploymentCondition(newStatus, *condition)
 		case deployapi.DeploymentStatusRunning:
-			msg := fmt.Sprintf("Replication controller %q is progressing", latestRC.Name)
-			condition := deployutil.NewDeploymentCondition(deployapi.DeploymentProgressing, kapi.ConditionTrue, deployutil.ReplicationControllerUpdatedReason, msg)
-			deployutil.SetDeploymentCondition(newStatus, *condition)
-		case deployapi.DeploymentStatusFailed:
-			if cond != nil && cond.Reason == deployutil.TimedOutReason {
-				break
+			if isProgressing {
+				deployutil.RemoveDeploymentCondition(newStatus, deployapi.DeploymentProgressing)
+				msg := fmt.Sprintf("Replication controller %q is progressing", latestRC.Name)
+				condition := deployutil.NewDeploymentCondition(deployapi.DeploymentProgressing, kapi.ConditionTrue, deployutil.ReplicationControllerUpdatedReason, msg)
+				// TODO: Right now, we use lastTransitionTime for storing the last time we had any progress instead
+				// of the last time the condition transitioned to a new status. We should probably change that.
+				deployutil.SetDeploymentCondition(newStatus, *condition)
 			}
+		case deployapi.DeploymentStatusFailed:
 			msg := fmt.Sprintf("Replication controller %q has failed progressing", latestRC.Name)
 			condition := deployutil.NewDeploymentCondition(deployapi.DeploymentProgressing, kapi.ConditionFalse, deployutil.TimedOutReason, msg)
 			deployutil.SetDeploymentCondition(newStatus, *condition)
 		case deployapi.DeploymentStatusComplete:
-			if cond != nil && cond.Reason == deployutil.NewRcAvailableReason {
-				break
-			}
 			msg := fmt.Sprintf("Replication controller %q has completed progressing", latestRC.Name)
 			condition := deployutil.NewDeploymentCondition(deployapi.DeploymentProgressing, kapi.ConditionTrue, deployutil.NewRcAvailableReason, msg)
 			deployutil.SetDeploymentCondition(newStatus, *condition)
 		}
+	} else if config.Status.LatestVersion == 0 && deployutil.HasImageChangeTrigger(&config) {
+		condition := deployutil.NewDeploymentCondition(deployapi.DeploymentProgressing, kapi.ConditionUnknown, "", "Waiting on an image update")
```

Comment:
We don't wait for config changes, the trigger controller bumps latestVersion as soon as it notices a config change. We have this message in `oc status` and I figured it would be more discoverable to have it here but I should have it in a separate PR.